source/CScriptEngine.cpp

```diff
@@ -708,12 +708,15 @@ namespace CLEO
                     if(customWorkDir != nullptr)
                         fsPath = ResolvePath(customWorkDir) / fsPath;
                     else
-                    {
-                        if (Filepath_Root.compare(GetWorkDir()) != 0) // ModLoader support: do not expand game dir relative paths
-                        {
-                            fsPath = GetWorkDir() / fsPath;
-                        }
-                    }
+                        fsPath = GetWorkDir() / fsPath;
+
+                    auto resolved = FS::weakly_canonical(fsPath).string();
+
+                    // ModLoader support: do not expand game dir relative paths
+                    if (resolved.find(Filepath_Root) == 0)
+                        return FS::relative(resolved, Filepath_Root).string();
+                    else
+                        return resolved;

```

Comment:
``suggestion
                return resolved.string();
```

Reply:
necessary? resolved is string already

Reply:
Oh ok. I confussed it with `FS::path resolved;` below.